nixos/lib/utils.nix

```diff
@@ -194,6 +194,21 @@ rec {
       (( ! $inherit_errexit_enabled )) && shopt -u inherit_errexit
     '';
 
+  /* Remove packages of packagesToRemove from packages, based on their names
```

Comment:
One reason I was reluctant to move this to some more official place like `lib` is that the function is terrible – both due to its quadratic complexity and reliance on package names (which is an ugly hack, IMO, just see what pains `nix-env` causes us).

So if we wanted to make it global like this, we would need to at least adjust the comment to warn against the usage of the function.

Reply:
wouldn't it make more sense to just specify a list of names instead of a list which gets implicitly mapped to gather the names to remove, and make the mapping, if any happens, explicit in the end user code? and maybe add some transition code which can handle both cases and use it in the gnome case...?

Reply:
> wouldn't it make more sense to just specify a list of names instead of a list which gets implicitly mapped to gather the names to remove, and make the mapping, if any happens, explicit in the end user code?

No because if a new package gets added everyone would need to change their configs. This way it is easier if you just want to remove one package without rebuilding the world.

> imo the name does not fit perfectly, because it doesn't take a list of strings, but a list of derivations, even tho it only uses the names..

It wouldn't work with strings when a package gets updated to an unstable version. You would need to include `package` and `package-unstable`.

> Or split it into two functions, one which just takes a list of names, and another one which adds the terrible obscure glue to first map derivations to their names.

Feel free to open a PR for that.